src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -632,11 +633,22 @@ internal override void ForceComplete(SourceLocation? locationOpt, CancellationTo
 
                             EnsureFieldDefinitionsNoted();
 
-                            // We've completed all members, so we're ready for the PointedAtManagedTypeChecks;
-                            // proceed to the next iteration.
-                            state.NotePartComplete(CompletionPart.MembersCompleted);
-                            break;
+
+                            if (state.NotePartComplete(CompletionPart.MembersCompletedChecksStarted))
```

Comment:
nit: The name of the completion flag is odd, as it suggests it comes in play when members are completed, when in fact it comes before setting `CompletionPart.MembersCompleted`. Consider `MembersChecksStarted` or `MembersCompletionChecksStarted`.

Reply:
> we start the checks we perform after members are completed.

There is a rub here. I that the `MembersCompleted` completion part to indicate when the members are completed. The checks we're adding come just before members are completed.

Reply:
> The checks we're adding come just before members are completed.

The members are all completed when these checks are run. We don't publish that fact yet, because we want to run some checks now that they are completed, but they are completed.

Reply:
Yes, I get that we have the members at hand. But we can't say they are completed until we've marked that part as complete. That's the definition of `CompletionPart.MembersCompleted`.
Perhaps `CompletionPart.StartValidatingMembers`?